server/src/main/java/org/elasticsearch/node/NodeConstruction.java

```diff
@@ -1324,17 +1314,26 @@ private Supplier<DocumentParsingObserver> getDocumentParsingObserverSupplier() {
      *
      * @see Node#BREAKER_TYPE_KEY
      */
-    private static CircuitBreakerService createCircuitBreakerService(
-        Settings settings,
-        List<BreakerSettings> breakerSettings,
-        ClusterSettings clusterSettings
-    ) {
+    private CircuitBreakerService createCircuitBreakerService(Settings settings, ClusterSettings clusterSettings) {
+        List<BreakerSettings> breakerSettings = pluginsService.filterPlugins(CircuitBreakerPlugin.class)
+            .map(plugin -> plugin.getCircuitBreaker(settings))
+            .toList();
+
         String type = Node.BREAKER_TYPE_KEY.get(settings);
-        return switch (type) {
+        CircuitBreakerService circuitBreakerService = switch (type) {
             case "hierarchy" -> new HierarchyCircuitBreakerService(settings, breakerSettings, clusterSettings);
             case "none" -> new NoneCircuitBreakerService();
             default -> throw new IllegalArgumentException("Unknown circuit breaker type [" + type + "]");
         };
+        resourcesToClose.add(circuitBreakerService);
+        modules.bindToInstance(CircuitBreakerService.class, circuitBreakerService);
+
+        pluginsService.filterPlugins(CircuitBreakerPlugin.class).forEach(plugin -> {
```

Comment:
nit: this filter on plugins is done twice. Can this be consolidated to collect the relevant plugin instances at the beginning of the method and shared?